tokio-trace/tokio-trace-core/src/callsite.rs

```diff
@@ -24,6 +24,27 @@ struct Registry {
     dispatchers: Vec<dispatcher::Registrar>,
 }
 
+impl Registry {
+    fn establish_interest(&self, callsite: &'static Callsite) {
+        let meta = callsite.metadata();
+
+        self.dispatchers.iter().for_each(|registrar| {
+            if let Some(interest) = registrar.try_register(meta) {
+                callsite.add_interest(interest)
+            }
+        });
+    }
+
+    fn reestablish(&mut self) {
+        self.dispatchers.retain(|registrar| registrar.is_alive());
```

Comment:
Nit/TIOLI: I think this can just be 
```suggestion
        self.dispatchers.retain(Registrar::is_alive);
```
not super important, though, I'm fine with it as-is.